rust/private/rustc.bzl

```diff
@@ -178,10 +189,43 @@ def rustc_compile_action(
     if ctx.file.out_dir_tar:
         compile_inputs.append(ctx.file.out_dir_tar)
 
+    rpaths = _compute_rpaths(toolchain, output_dir, depinfo)
+
+    if not hasattr(ctx.fragments.cpp, "linkopts"):
+        fail("Bazel is too old, please use at least Bazel 0.18.0")
+    elif versions.is_at_least("0.18.0", BAZEL_VERSION):
+        user_link_flags = ctx.fragments.cpp.linkopts
+    else:
+        user_link_flags = depset(ctx.fragments.cpp.linkopts)
+
     # Paths to cc (for linker) and ar
     cpp_fragment = ctx.host_fragments.cpp
-    cc = cpp_fragment.compiler_executable
-    ar = cpp_fragment.ar_executable
+    cc_toolchain = find_cpp_toolchain(ctx)
+    feature_configuration = cc_common.configure_features(
```

Comment:
Where is `cc_common` defined and how does it tie into cc_toolchain?

Reply:
cc_common is currently builtin, the list of all of the builtins is here: https://docs.bazel.build/versions/master/skylark/lib/skylark-overview.html.